### ISSUE TITLE
AllReduce info.proto returns 0 (LL protocol) despite both setting env variable to SIMPLE / Tuning.cc protoEnable array disabling LL/LL128

### ISSUE THREAD

**Issue by OckermanSethGVSU** [NVIDIA/nccl#1246]
I am facing an odd issue where I am seeing the LL protocol being used despite setting the protocol to SIMPLE in my nccl.conf file. To confirm this, I added a basic print statement to `collectives.cc` to get the protocol (shown below) and then profiled my execution using NSIGHT. My debug file lists SIMPLE as the protocol set by the  env. but NSIGHT and the info.protocol return do not agree. 

```
if (info.protocol != -1){
   
      std::cout << ncclProtoToString(info.protocol) << std::endl;
    
  }
``` 

In an attempt to fix this, I went into [`tuning.cc` line 270](https://github.com/NVIDIA/nccl/blob/master/src/graph/tuning.cc#L270) and set the value to `int protoEnable[NCCL_NUM_PROTOCOLS] = { 0, 0, 1 };` which in theory should have disabled the LL and LL128 protocls. When that did not work, I then went into [`tuning.cc`'s for loop on line 303](https://github.com/NVIDIA/nccl/blob/master/src/graph/tuning.cc#L303) and manually used if statements to set only `NCCL_PROTO_SIMPLE pEnable = 1`. That still did not work...

I am using NCCL version 2.21.5+cuda12.4. If I just misunderstood how the source code works I apologize in advance.

**Comment by OckermanSethGVSU** [comment]
Is that true also of the info call in AllReduce itself? 

I added a print statement to AllReduce that prints "info.proto"  and it returns 0 for the LL proto.

**Comment by sjeaugey** [comment]
`info` should be correct. Did you try to set `NCCL_PROTO=SIMPLE` in the environment? That's what will eventually set the proto enable flags.

**Comment by OckermanSethGVSU** [comment]
Yes. I have both an .nccl_conf file and when that didn't work I also tried exporting nccl_proto=Simple (with correct caps) in the env. My debug file reports that the proto is set to SIMPLE by the env.

**Comment by OckermanSethGVSU** [comment]
Could you point me to where the NCCL_PROTO is finally selected? I can find where different protos are enabled / disabled, but I cannot find the final selection.
